persistence/build.gradle

```diff
@@ -6,6 +6,6 @@ dependencies {
     compile hibernate_java8
     compile postgres_jdbc_driver
     testCompile spring_test
-    testCompile hsqldb
+    compile hsqldb
```

Comment:
It's incorrect. It means you have beans configured incorrectly or in another context. This dependency MUST be testCompile. Check spring beans implementation.